static/js/src/contributions.js

```diff
@@ -143,29 +143,22 @@ function sendContributionFormAnalytics() {
   const inputs = document.querySelectorAll(".p-slider__input");
   const total = document.querySelector(".js-total-amount").innerText;
 
+  let dataLayerObject = {
+    event: "contribution",
+    total: total
+  };
+
   // add the individual items
   inputs.forEach((amountElement) => {
     const name = amountElement.parentNode.parentNode.id;
     let value = amountElement.value || 0;
 
     if (value > 0) {
-      dataLayer.push({
-        event: "GAEvent",
-        eventCategory: "Contributions",
-        eventAction: "Item",
-        eventLabel: name,
-        eventValue: value,
-      });
+      dataLayerObject.name =  value;
     }
   });
```

Comment:
Isn't this pushing only the last object in `inputs` to the `dataLayer` since `dataLayerObject.name` is being overwritten on every pass?

`dataLayerObject[name] = value;` could work, as it adds the values for each unique key, but it would also overwrite values with the same key name

Reply:
maybe i'm wrong but i suppose each iteration of the forEach will give a different value to "name" and the dataLayerObject.name would be replaced with the value of "name". so dataLayerObject will end up with a new key-value pair at each iteration.

Reply:
Right now the iteration is directly referencing the `dataLayerObject.name` attribute, so each pass rewrites the value on the same attribute rather than storing the keys